### PR TITLE
Amending pipeline support (#145)

### PR DESCRIPTION
**Pull request by olivergondza**
Some more tweaks before this is merged:

- Move pipeline support to dedicated package
- Null safety fixes
- Make the workflow dependency optional

@witokondoria, could you give it a try so we can release it?

### CODE REVIEW

File: src/main/java/jenkins/plugins/openstack/pipeline/SimplifiedServer.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        JCloudsCloud jcl = JCloudsCloud.getByName(cloud);
+        JCloudsSlaveTemplate t = jcl.getTemplate(template);
+        if (t == null) throw new IllegalArgumentException("Invalid template: " + template);
+
+        this.srv = t.provision(jcl, serverscope);
+    }
+
+    @Whitelisted
+    public void destroy() {
+        if (srv == null) return; // Already terminated.
```
Comment (witokondoria) [inline src/main/java/jenkins/plugins/openstack/pipeline/SimplifiedServer.java]: I'm a big fan of [braces on ifs](http://wiki.c2.com/?AlwaysUseBracesOnIfThen)

File: src/test/java/jenkins/plugins/openstack/jenkins/plugins/openstack/pipeline/OpenstackStepTest.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
+package jenkins.plugins.openstack.jenkins.plugins.openstack.pipeline;
```
Comment (witokondoria) [inline src/test/java/jenkins/plugins/openstack/jenkins/plugins/openstack/pipeline/OpenstackStepTest.java]: This package seems wrecked

Comment (olivergondza) [inline src/test/java/jenkins/plugins/openstack/jenkins/plugins/openstack/pipeline/OpenstackStepTest.java]: Uh, right.

**Comment by olivergondza** [comment]
Thanks for verifying!

**Pull request merged**